crates/extensions/tedge_gen_mapper/src/js_filter.rs

```diff
@@ -0,0 +1,360 @@
+use crate::js_runtime::JsRuntime;
+use crate::pipeline;
+use crate::pipeline::DateTime;
+use crate::pipeline::FilterError;
+use crate::pipeline::Message;
+use anyhow::Context;
+use rquickjs::Ctx;
+use rquickjs::FromJs;
+use rquickjs::IntoJs;
+use rquickjs::Value;
+use std::path::Path;
+use std::path::PathBuf;
+use tracing::debug;
+
+#[derive(Clone)]
+pub struct JsFilter {
+    pub module_name: String,
+    pub path: PathBuf,
+    pub config: JsonValue,
+    pub tick_every_seconds: u64,
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct JsonValue(serde_json::Value);
+
+impl JsFilter {
+    pub fn new(pipeline: PathBuf, index: usize, path: PathBuf) -> Self {
+        let module_name = format!("{}|{}|{}", pipeline.display(), index, path.display());
```

Comment:
The `index` is used to distinguish instances of javascript module used by the same pipeline on different stages.

- I still have to add a test for that case.
- The current full name is a bit large. What about using a hash?